src/lib/server.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+const net = require('net');
+const logger = require('./logger');
+const faker = require('faker');
+
+const app = net.createServer();
+let clients = [];
+
+const parseCommand = (message, socket) => {
+  if (!message.startsWith('@')) {
+    return false;
+  }
+
+  const parsedMessage = message.split(' ');
+  const command = parsedMessage[0];
+  logger.log(logger.INFO, `parsing a command ${command}`);
+
+  switch (command) {
+    case '@list': {
+      const clientNames = clients.map(client => client.name).join('\n');
+      socket.write(`${clientNames}\n`);
+      break;
+    }
+    // where new commands go!
+    case '@nickname': {
+      const newName = parsedMessage[1];
+      // need to update name in clients
+
+      socket.write(`Your new name is ${newName}`);
+      break;
+    }
+    case '@quit': {
+      socket.close();
```

Comment:
socket.end() is what you're trying to do. socket.close() isn't a function and breaks the code.